controllers/services/auth/auth_controller_actions.go

```diff
@@ -0,0 +1,231 @@
+				if !slices.Contains(ai.Spec.AdminGroups, e) {
+					ai.Spec.AdminGroups = append(ai.Spec.AdminGroups, strings.TrimSpace(e))
+					newGroups = true
+				}
+			}
+		}
+
+		allowedg := foundGroups["allowedGroups"]
+		if len(allowedg) != 0 {
+			for _, e := range strings.Split(allowedg, ",") {
+				// If the group is already in the list then ignore it
+				if !slices.Contains(ai.Spec.AllowedGroups, e) {
+					ai.Spec.AllowedGroups = append(ai.Spec.AllowedGroups, strings.TrimSpace(e))
+					newGroups = true
+				}
+			}
+		}
+	}
+		
+	if len(ai.Spec.AdminGroups) == 0 {
+		// adminGroups is empty so add the default
+		ai.Spec.AdminGroups = append(ai.Spec.AdminGroups, dashboard.GetAdminGroup())
+		newGroups = true
+	}
+
+	if len(ai.Spec.AllowedGroups) == 0 {
+		// allowedGroups is empty so add the default
+		ai.Spec.AllowedGroups = append(ai.Spec.AllowedGroups, "system:authenticated")
+		newGroups = true
+	}
+
+	// only update if we found a new group in the list
+	if newGroups {
+		err := rr.Client.Update(ctx, ai)
+		if err != nil {
+			return errors.New("error adding groups to Auth CR")
+		}
+	}
+
+	return nil
+}
+
+func bindRole(ctx context.Context, rr *odhtypes.ReconciliationRequest, groups []string, roleBindingName string, roleName string) error {
+	var groupsToBind []rbacv1.Subject
+	for _, e := range groups {
+		rs := rbacv1.Subject{
+			Kind:     "Group",
+			APIGroup: "rbac.authorization.k8s.io",
+			Name:     e,
+		}
+		groupsToBind = append(groupsToBind, rs)
+	}
+
+	rb := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleBindingName,
+			Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+		},
+		Subjects: groupsToBind,
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     roleName,
+		},
+	}
+	err := rr.Client.Update(ctx, rb)
+	if err != nil {
+		return errors.New("error creating RoleBinding for group")
+	}
+
+	return nil
+}
+
+func removeOldPermissions(ctx context.Context, rr *odhtypes.ReconciliationRequest, groups []string, roleBindingName string, roleName string) error {
+	roleBinding := &rbacv1.RoleBinding{}
+	if err := rr.Client.Get(ctx, client.ObjectKey{
+		Name:      roleBindingName,
+		Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+	}, roleBinding); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	var groupsToRemove []rbacv1.Subject
+	for _, s := range roleBinding.Subjects {
+		var found = false
+		for _, g := range groups {
+			if s.Name == g {
+				found = true
+			}
+		}
+		if !found {
+			groupsToRemove = append(groupsToRemove, s)
+		}
+	}
+
+	if len(groupsToRemove) > 0 {
+		rb := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      roleBindingName,
+				Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+			},
+			Subjects: groupsToRemove,
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "Role",
+				Name:     roleName,
+			},
+		}
+
+		err := rr.Client.Delete(ctx, rb)
```

Comment:
wonder if here you meant `Update` ?  if `Delete` is the action you want to perform, then you don't need to fill all the various properties of the object (`Subjects`, `RoleRef`, ect)

Reply:
:+1: